src/context-parser.js

```diff
@@ -180,6 +180,14 @@ FastParser.prototype.walk = function(i, input, endsWithEOF) {
             if(this.tags[0].toLowerCase() === this.tags[1].toLowerCase()) {
                 reconsume = 0;  /* see 12.2.4.13 - switch state for the following case, otherwise, reconsume. */
                 this.matchEndTagWithStartTag(symbol);
+                /* 
+                  After matchEndTagWithStartTag (with start tag name == end tag name), 
+                  the state will be transition back to DATA. 
```

Comment:
this is true only during https://github.com/yahoo/context-parser/pull/38/files#diff-453c94db9d68bf214cb0505c0e0791b5R245

expect to move `this.tags[0] = '';` to there